### PR TITLE
cmd/govim: fix panic when using custom hover opts "col" / "line"

### PR DESCRIPTION
**Pull request by leitzler**
Setting either ExperimentalCursorTriggeredHoverPopupOptions or
ExperimentalMouseTriggeredHoverPopupOptions fields "col" or "line"
did result in a panic.

The value was parsed as a raw json message, but since it is already
parsed once before it paniced. We now type assert the value instead.

Updates #1012

### CODE REVIEW

File: cmd/govim/hover.go [hunk 1]
```diff
@@ -113,16 +115,11 @@ func (v *vimstate) showHover(posExpr string, opts map[string]interface{}, userOp
 			opts[k] = v
 		}
 		var line, col int64
-		var err error
-		if lv, ok := opts["line"]; ok {
-			if line, err = rawToInt(lv); err != nil {
-				return nil, fmt.Errorf("failed to parse line option: %v", err)
-			}
+		if lv, ok := opts["line"].(float64); ok {
+			line = int64(lv)
```
Comment (myitcv) [inline cmd/govim/hover.go:119]: I think we probably want `math.Round` here? Otherwise it truncates towards zero.

Alternatively we could flip the decoding of the opts to use [`UseNumber`](https://pkg.go.dev/encoding/json#Decoder.UseNumber) (which might actually be a nice change, separate from this one)

**Review by myitcv — APPROVED** [review]
LGTM modulo one suggestion. 

Could also add a small test to cover this edge case?

File: cmd/govim/hover.go (hunk 1 above)
Comment (leitzler) [reply cmd/govim/hover.go:119]: Interesting, would it ever be an issue to not use `math.Round` if the user always specify integers?

I wouldn't expect anyone to ever specify anything outside like, the range -10 to 10. Since it is just for line and column I would be surprised if anyone tried to set "line 1.7" :)

Comment (myitcv) [reply cmd/govim/hover.go:119]: The string value that represents the number gets converted to a `float`. There is no guarantee that the integer number represented by the string can be represented precisely as a float. Hence the truncation towards zero could cause an off by one if the values ends up being x.9999999 for example.

I'd prefer to use `UseNumber()` across the board for this reason, but for now `math.Round` is a sufficient workaround.

Comment (leitzler) [reply cmd/govim/hover.go:119]: I see! It wasn't an objection, was only out of curiosity :)
